### PR TITLE
Run MirrorIT on Windows

### PR DESCRIPTION
**Pull request by infeo**
This PR:
* updates the windows jfuse implementation to the same state as the mac and linux impl (mount, rund and unmount without errors or exceptions)
* activates the windows mirror integration tests on the CI

Regarding 25603a02f7cdf2e1189a938cec49a196ae3adddb:
Due to the first [failed CI run](https://github.com/cryptomator/jfuse/actions/runs/1999695202) with unclear error message, i thought maybe we have to mount into a directory and prepared everything in the integration test for it. Because I'm messy i forgot to set the property in the ci script, but the tests were successful nonetheless. 🤷

### CODE REVIEW

File: jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/PosixMirrorFileSystem.java [hunk 1]
```diff
@@ -38,6 +38,8 @@ public static void main(String[] args) {
 			} else {
 				LOG.error("Failed to mount to {}. Exit code: {}", mountPoint, result);
 			}
+			System.out.println("Enter a anything to unmount...");
```
Comment (overheadhunter) [inline jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/PosixMirrorFileSystem.java:41]:
```suggestion
			LOG.info("Enter a anything to unmount...");
```

File: jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/WindowsMirrorFileSystem.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public static void main(String[] args) {
 			} else {
 				LOG.error("Failed to mount to {}. Exit code: {}", mountPoint, result);
 			}
+			System.out.println("Enter a anything to unmount...");
```
Comment (overheadhunter) [inline jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/WindowsMirrorFileSystem.java:40]:
```suggestion
			LOG.info("Enter a anything to unmount...");
```

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java [hunk 3]
```diff
@@ -42,17 +46,27 @@ public void setup(@TempDir Path tmpDir) throws IOException, TimeoutException {
...
+		flags.add("-s");
 		if (OS.WINDOWS.isCurrentOs()) {
-			mirror = Path.of("M:");
+			//Idea: read from system property! in build script we can set it via "-DmyProp=.."
+			var ciMountPath = System.getProperty(WIN_MOUNT_DIR_PROP);
+			if( ciMountPath != null) {
+				mirror = Path.of(ciMountPath).resolve("mnt");
+			} else {
+				mirror = Path.of("M:\\");
+			}
```
Comment (overheadhunter) [inline jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java:58]: this doesn't get set during CI build,  does it? Is it required? Maybe we can use the "find any free drive letter" function from Cryptomator instead?

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java [hunk 4]
```diff
@@ -42,17 +46,27 @@ public void setup(@TempDir Path tmpDir) throws IOException, TimeoutException {
...
 		Files.createDirectories(orig);
 		AbstractMirrorFileSystem fs;
+		List<String> flags = new ArrayList<>();
+		flags.add("-s");
 		if (OS.WINDOWS.isCurrentOs()) {
-			mirror = Path.of("M:");
+			//Idea: read from system property! in build script we can set it via "-DmyProp=.."
+			var ciMountPath = System.getProperty(WIN_MOUNT_DIR_PROP);
+			if( ciMountPath != null) {
+				mirror = Path.of(ciMountPath).resolve("mnt");
```
Comment (overheadhunter) [inline jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java:55]: Oh right, WinFSP supports mounting to (non-existing) paths now! In this case, remove the "drive letter" thing from this test alltogether

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java [hunk 5]
```diff
@@ -207,4 +224,10 @@ private int write(MemoryAddress path, MemoryAddress buf, long size, long offset,
 		}
 	}
 
+	@Override
+	public void close() throws TimeoutException {
+		fuseExit();
```
Comment (overheadhunter) [inline jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:229]: please mark with TODO for now, as we're not sure about the final API for unmounting

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java [hunk 6]
```diff
@@ -48,13 +67,11 @@ protected int fuseMain(int argc, MemorySegment argv) {
...
-//			var ctx = fuse_context.ofAddress(fuse_h.fuse_get_context(), scope);
-//			var fusePtr = fuse_context.fuse$get(ctx);
-//			fuse_h.fuse_exit(fusePtr);
-//		}
-//	}
+	private void fuseExit() {
+		if (fuseHandle != null) {
+			fuse_h.fuse_exit(fuseHandle);
+		}
+	}
```
Comment (overheadhunter) [inline jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:74]: please mark with TODO for now, as we're not sure about the final API for unmounting

**Review by overheadhunter — CHANGES_REQUESTED** [review]

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java (hunk 5 above)
Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:229]: Does the intellij package also provide some sort of `@Incubating` annotation? If so, we could use this one instead.

Comment (overheadhunter) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:229]: `@Experimental`, but this is rather for APIs not for implementation....

Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:229]: `TODO` it is then!

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java (hunk 4 above)
Comment (infeo) [reply jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java]: Fixed in [dd0495e](https://github.com/cryptomator/jfuse/pull/2/commits/dd0495e1993ae85f18275273e4ac65f958c1b36c)

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java (hunk 3 above)
Comment (infeo) [reply jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java]: see https://github.com/cryptomator/jfuse/pull/2#discussion_r829879236

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java (hunk 6 above)
Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java]: Fixed in [b22aac6](https://github.com/cryptomator/jfuse/pull/2/commits/b22aac6097be1d293f8c5d146e37cb89e74d4ec5)

File: jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java (hunk 5 above)
Comment (infeo) [reply jfuse-win-amd64/src/main/java/org/cryptomator/jfuse/win/amd64/FuseImpl.java:234]: Fixed in [b22aac6](https://github.com/cryptomator/jfuse/pull/2/commits/b22aac6097be1d293f8c5d146e37cb89e74d4ec5)

File: jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/PosixMirrorFileSystem.java (hunk 1 above)
Comment (infeo) [reply jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/PosixMirrorFileSystem.java]: Fixed in [b22aac6](https://github.com/cryptomator/jfuse/pull/2/commits/b22aac6097be1d293f8c5d146e37cb89e74d4ec5)

File: jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/WindowsMirrorFileSystem.java (hunk 2 above)
Comment (infeo) [reply jfuse-examples/src/main/java/org/cryptomator/jfuse/examples/WindowsMirrorFileSystem.java]: Fixed in [b22aac6](https://github.com/cryptomator/jfuse/pull/2/commits/b22aac6097be1d293f8c5d146e37cb89e74d4ec5)

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java [hunk 7]
```diff
@@ -26,12 +26,16 @@
...
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class MirrorIT {
 
+	private static final String WIN_MOUNT_DIR_PROP = "jfuse.win.integrationMountDir";
+
```
Comment (overheadhunter) [inline jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java:38]: no longer required

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -48,9 +48,30 @@ jobs:
...
+          name: coverage-win-amd64
+          path: jfuse-tests/target/site/jacoco-aggregate/jacoco.xml
+          retention-days: 3
+
   sonarcloud:
     name: Run SonarCloud Analysis
-    needs: [linux-amd64, mac]
+    needs: [linux-amd64, mac, win-amd64]
     runs-on: ubuntu-latest
     if: "!contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]')"
```
Comment (overheadhunter) [inline .github/workflows/build.yml:76]: can be removed, since this job depends on other jobs that have this condition already

Comment (overheadhunter) [reply .github/workflows/build.yml:76]: (I'm aware this is kind of unrelated to this PR, so feel free to ignore this and do it on develop instead)

**Review by overheadhunter — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 8 above)
Comment (overheadhunter) [reply .github/workflows/build.yml:76]: ok this is very ambiguous. This comment refers to line 76

File: jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java (hunk 7 above)
Comment (infeo) [reply jfuse-tests/src/test/java/org/cryptomator/jfuse/tests/MirrorIT.java]: Fixed in https://github.com/cryptomator/jfuse/pull/2/commits/cb670365b986c16d72ca3af05369d40dddf62ee6

**Review by overheadhunter — APPROVED** [review]

**Pull request merged**